File: Chapter9_Example/src/main/java/MessageBus.java
public class MessageBus {
    private final IBus _bus;

    public MessageBus(IBus _bus) {
        this._bus = _bus;
    }

    public void sendEmailChangedMessage(int userId, String newEmail) {
        _bus.send("Type : USER EMAIL CHANGED; Id : " + userId + "; NewEmail : " + newEmail);
    }
}

File: unit-testing/src/main/kotlin/book/chapter7/_refactored2/MessageBus.kt
package book.chapter7._refactored2

class MessageBus {

    private val bus: IBus
        get() {
            TODO()
        }

    fun sendEmailChangedMessage(userId: Int, newEmail: String) {
        bus.send("Subject: USER; Type: EMAIL CHANGED; Id: $userId; NewEmail: $newEmail")
    }

}